scrape_repo_information.py

```diff
@@ -0,0 +1,78 @@
+import urllib.request
+import pdb
+import xml.etree.ElementTree as xml
+import argparse, urllib.request
+import json
+import collections
+import time
+import urllib
+import os
+"""
+Script to scrape text from GitHub and into JSON format for all language pairs in Apertium.
+Stem counter by sushain
+"""
+
+CLIENT_ID = os.environ['GITHUB_CLIENT_ID']
+CLIENT_SECRET = os.environ['GITHUB_SECRET_CLIENT_ID']
+
+pairs = []
+Pair = collections.namedtuple('Pair', 'lg1 lg2 last_updated created direction repo stems')
+types = ["trunk", "nursery", "incubator", "staging"]
+params = urllib.parse.urlencode(dict({'client_id': CLIENT_ID, 'client_secret': CLIENT_SECRET}))
+
+def print_info(uri):
+    returned = get_info(uri)
+    return returned
+
+def get_info(uri):
+    try:
+        dictX = str((urllib.request.urlopen(uri)).read(), 'utf-8')
+        tree = xml.fromstring(dictX)
+    except:
+        return -1 # FIXME: error handling
+
+    return len(tree.findall("*[@id='main']/e//l"))
+
+for x in types:
+    html_url = "https://api.github.com/repos/apertium/apertium-%s/contents?" % x
+    html_data = urllib.request.urlopen(html_url + params)
+    html = json.loads(html_data.read())
+    for i in html:
+        if "apertium" in i["name"]:
+            try:
+                repo_name = x
+                direction = ""
+                lang_pair_name = i["name"]
+                #getting names
+                lg1 = lang_pair_name.split('-')[1]
+                lg2 = lang_pair_name.split('-')[2]
+                #getting into repository
+                link = "https://api.github.com/repos/apertium/%s/contents?" % lang_pair_name
+                repo_json = json.loads(urllib.request.urlopen(link + params).read())
+                for el in repo_json:
+                    if el["name"] == "modes.xml":
+                        download_url = el["download_url"]
+                        html_for_blob = urllib.request.urlopen(download_url).read()
+                        if lg1+'-'+lg2 in str(html_for_blob):
+                            direction += "<"
+                        if lg2+'-'+lg1 in str(html_for_blob):
+                            direction += ">"
+                    elif ".dix" in el["name"]:
```

Comment:
Unfortunately, this will result in dictionaries aside from the main bidix, e.g. a postdix, getting picked up. There are also monodix on old pairs which shouldn't be picked up here.